modules/core/src/smithy4s/http/HttpUnaryServerRouter.scala

```diff
@@ -22,91 +22,181 @@ import smithy4s.server.UnaryServerCodecs
 import smithy4s.capability.MonadThrowLike
 import scala.annotation.nowarn
 
+// scalafmt: {maxColumn = 120}
 object HttpUnaryServerRouter {
 
+  /**
+    * Creates a router in the form isomorphic to `Kleisli[OptionT[F, *], Request, Response]`, which
+    * is useful for integration with frameworks such as Http4s.
+    */
   def apply[Alg[_[_, _, _, _, _]], F[_], Request, Response](
       service: smithy4s.Service[Alg]
   )(
       impl: service.Impl[F],
       makeServerCodecs: UnaryServerCodecs.Make[F, Request, Response],
       endpointMiddleware: Endpoint.Middleware[Request => F[Response]],
       getMethod: Request => HttpMethod,
-      getPathSegments: Request => IndexedSeq[String],
+      getUri: Request => HttpUri,
       addDecodedPathParams: (Request, PathParams) => Request
   )(implicit F: MonadThrowLike[F]): Request => Option[F[Response]] = {
-    val router =
-      new HttpUnaryServerRouter[Alg, service.Operation, F, Request, Response](
-        service,
-        service.toPolyFunction[smithy4s.kinds.Kind1[F]#toKind5](impl),
-        makeServerCodecs,
-        endpointMiddleware,
-        getMethod,
-        getPathSegments,
-        addDecodedPathParams
-      )
-    router.route(_)
+    new KleisliRouter[Alg, service.Operation, F, Request, Response](
+      service,
+      service.toPolyFunction[smithy4s.kinds.Kind1[F]#toKind5](impl),
+      makeServerCodecs,
+      endpointMiddleware,
+      getMethod,
+      getUri,
+      addDecodedPathParams
+    )
   }
 
-}
+  /**
+    * Creates a router in the form of a partial function, which is useful for integration with frameworks
+    * such as Play.
+    */
+  def partialFunction[Alg[_[_, _, _, _, _]], F[_], RequestHead, Request, Response](
```

Comment:
sometimes the simplest solution turns out to be the one to use

good thinking, it sucks that we sort of have duplicated router logic in those two classes, but it will make usage easier during integration, that's for sure

Reply:
that's mostly due to `RequestHead` being a distinct type from `Request`, right? I suppose this kind of duplication is just inevitable in cases like this.

Reply:
yes :/

and the nature of the partial function means that we have to run the logic twice, once to see if it should be applied, and the second time around to actually perform the routing